File: backend/src/main/java/co/arthur/petshop/petshop_api/entity/Pet.java
package co.arthur.petshop.petshop_api.entity;

import co.arthur.petshop.petshop_api.enums.EspecieEnum;
import co.arthur.petshop.petshop_api.enums.PorteEnum;
import co.arthur.petshop.petshop_api.enums.SexoEnum;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.util.Date;

@Entity
@Table(name = "pet")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;
    @Enumerated(EnumType.STRING)
    private EspecieEnum especie;
    @ManyToOne
    @JoinColumn(name = "raca_id")
    private Raca raca;
    @Enumerated(EnumType.STRING)
    private SexoEnum sexo;
    @Enumerated(EnumType.STRING)
    private PorteEnum porte;
    private LocalDate dataNascimento;
    private String observacao;
    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @CreationTimestamp
    private Date dataCadastro;
    @UpdateTimestamp
    private Date dataAtualizacao;
    @Column(nullable = false)
    private Boolean ativo = true;

}

File: frontend/src/modules/pet-edicao/data/pet.ts
import { gql } from 'graphql-request'

export const PET_QUERY = gql`
  query pet($id: ID!) {
    pet(id: $id) {
      id
      nome
      especie
      raca {
        id
        nome
      }
      sexo
      porte
      cliente {
        id
        nome
      }
      observacao
      dataNascimento
    }
  }
`

export const RACAS_QUERY = gql`
  query racas($especie: EspecieEnum!) {
    racas(especie: $especie) {
      id
      nome
      especie
    }
  }
`

export const SALVAR_PET_MUTATION = gql`
  mutation salvarPet($filterInput: PetInput!) {
    salvarPet(filterInput: $filterInput) {
      id
      nome
      especie
      raca {
        id
        nome
      }
      sexo
      porte
      cliente {
        id
        nome
      }
      observacao
      dataNascimento
    }
  }
`